### DISCUSSION TITLE
Update Node version

### DISCUSSION THREAD

**Discussion by lucpotage** [thebaselab/codeapp#354] [Ideas]
Thanks for your great app. It's the only app that makes me feel I can develop on iPad.

Unfortunately, after initiating a project and fetch dependencies, I had an error related to Node version and was not able to continue.

It's currently 12.19 in Code app whereas the latest Node LTS is currently 16.14. Do you think you could keep the Node version up-to-date?

**Comment by luni-moon** [comment]
Refer to issue #328

**Comment by bummoblizard** [comment]
Hey. It's being worked on https://github.com/nodejs-mobile/nodejs-mobile. I'm pretty sure it's doable but I can't tell when it will be done.

**Comment by bummoblizard** [comment]
It has been updated to Node 16 in the latest commit.

**Comment by wtpalexander** [comment]
Great to see Node 16, thanks so much for this work! Any ideas when this might make it to the AppStore? TestFlight beta is full and had some trouble installing through Xcode with the group ID’s and so on.

**Reply by bummoblizard** [reply]
It should be available within this week.

**Reply by wtpalexander** [reply]
Awesome, looking forward to it! 😄

**Reply by luni-moon** [reply]
Thanks!

**Comment by wtpalexander** [comment]
Am currently trying out the new Node 16 runtime. Am trying to run `npm install` and looks like I’m not having any luck. I get two different outputs alternating between the two. The second looks promising but returns almost instantly with no apparent installation. Still on my quest to run Nuxt (built on Vue.js) on iPad! 

![43B703A8-F972-4E5C-AF53-9F87A5859222](https://user-images.githubusercontent.com/7461082/157157095-21d9d538-0ecb-4ff4-b5b8-6cf763f416b1.jpeg)

Running with verbose login yields this:
![70DE1103-0A5D-4E2C-A50E-36A3690887BF](https://user-images.githubusercontent.com/7461082/157157300-8a59f762-0000-4353-941f-3a4b14158beb.jpeg)

Running on iPad Pro 3rd Gen, iPad OS 15.2.1 

Any ideas @bummoblizard?

**Reply by bummoblizard** [reply]
Running node on iOS is a fragile process. If any modules contain native modules, or spawn subprocess, it will immediately fail (sometimes silently). I will try different combinations of modules versions to get Nuxt running on Code App.

**Reply by wtpalexander** [reply]
Ok that's a shame: so there's a possibility that this is just genuinely impossible with the current state of mobile Node?

**Reply by bummoblizard** [reply]
I set up a Vue project a while ago that can be run on Code App: https://github.com/bummoblizard/vue-archive
Maybe that could be used as a starting point.

**Reply by wtpalexander** [reply]
Interestingly I get the same errors as I do with Nuxt when trying your Vue repo. Not sure what the difference could be? I wonder if the AppStore build could be our difference?

**Reply by bummoblizard** [reply]
Deleting package-lock.json worked for me. Npm seems to spawn a sub process when fixing the older version package-lcok.json.
```
npm i
npm run demo
```
If `npm i` failed with `Program interrupted` or silently, you might need to run it again.

**Reply by bummoblizard** [reply]
Node 16 also fixed an issue of webpack middleware where previously it cannot watch file changes and recompile accordingly.

In `index.js`, this can be omitted:
``` js
watchOptions: {
    ignored: /.*/
  }
```
